scripts/code_style_rewrite_branch.py

```diff
@@ -0,0 +1,253 @@
+    """
+    Return the list of commits in revision_or_range.
+
+    If revision_or_range is a single revision, return it in a one-element
+    list. Otherwise return the list of commits in that range.
+
+    Return the list in order from oldest to newest.
+    """
+    result = subprocess.run(["git", "rev-list", "--no-walk", \
+            revision_or_range], stdout=subprocess.PIPE, check=False)
+    if result.returncode != 0:
+        print("Error getting revision list for '"+revision_or_range+"'.", \
+                file=sys.stderr)
+        return []
+
+    return list(reversed(str(result.stdout, "ascii").split()))
+
+def rewrite_branch_code_style(existing_branch: str, new_branch: str, \
+        codestyle_change_commit: str) -> bool:
+    """
+    Rewrite the given existing branch onto the given new branch on top of the
+    given commit that changes the code style.
+    """
+    # Create the new branch to rewrite onto at the code style change commit
+    result = subprocess.run(["git", "checkout", "--detach",
+                             codestyle_change_commit], \
+            check=False)
+    if result.returncode != 0:
+        print("Error checking out code style commit '" + \
+                codestyle_change_commit + "'.", file=sys.stderr)
+        return False
+    result = subprocess.run(["git", "checkout", "-b", new_branch], check=False)
+    if result.returncode != 0:
+        print("Error creating destination branch '"+new_branch+"'.", \
+                file=sys.stderr)
+        return False
+
+    # List all the commits between the commit before the codestyle change
+    # (where the branch diverged) and the tip of the branch.
+    for rev in list_revisions(codestyle_change_commit+"~.."+existing_branch):
+        if not restyle_commit_onto_current(rev):
+            delete_branch(new_branch)
+            return False
+
+    return True
+
+def main() -> int:
+    """
+    Main with command line arguments.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('existing_branch', \
+            help='branch to rewrite with code style changes')
+    parser.add_argument('destination_branch', \
+            help='name of new branch for rewritten history (must not be the' \
+            'same as existing_branch)')
+    parser.add_argument('code_style_commit', \
+            help='the commit containing the code style changes')
+
+    args = parser.parse_args()
+
+    # Check that the working directory is clean
+    if not working_directory_clean():
+        print("Error: Working directory is not clean.", file=sys.stderr)
+        return 1
+
+    # First rebase the branch on the commit preceding the code style change
+    rebased_existing_branch = "rebased-"+args.existing_branch
+    if not rebase_on(args.existing_branch, args.code_style_commit+"~", \
+            rebased_existing_branch):
+        return 1
+
+    # Then rewrite other commits to be code-styled
```

Comment:
If destination branch is not needed, we should simplify the logic and just apply the changes to the tempory rebased branch, append the new commits and check out in a detached stage.